homeassistant/components/metoffice/sensor.py

```diff
@@ -30,175 +34,189 @@
-        return f"{self._name} {SENSOR_TYPES[self._condition][0]}"
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return SENSOR_TYPES[self._type][2]
 
     @property
-    def state(self):
-        """Return the state of the sensor."""
-        if self._condition == "visibility_distance" and hasattr(
-            self.data.data, "visibility"
-        ):
-            return VISIBILITY_CLASSES.get(self.data.data.visibility.value)
-        if hasattr(self.data.data, self._condition):
-            variable = getattr(self.data.data, self._condition)
-            if self._condition == "weather":
-                return [
-                    k
-                    for k, v in CONDITION_CLASSES.items()
-                    if self.data.data.weather.value in v
-                ][0]
-            return variable.value
-        return None
+    def icon(self):
+        """Return the icon for the entity card."""
+        value = SENSOR_TYPES[self._type][3]
+        if self._type == "weather":
+            value = self.state
+            if value == "partlycloudy":
+                value = "partly-cloudy"
+            value = f"mdi:weather-{value}"
+
+        return value
 
     @property
-    def unit_of_measurement(self):
-        """Return the unit of measurement."""
-        return SENSOR_TYPES[self._condition][1]
+    def device_class(self):
+        """Return the device class of the sensor."""
+        return SENSOR_TYPES[self._type][1]
 
     @property
     def device_state_attributes(self):
         """Return the state attributes of the device."""
-        attr = {}
-        attr[ATTR_ATTRIBUTION] = ATTRIBUTION
-        attr[ATTR_LAST_UPDATE] = self.data.data.date
-        attr[ATTR_SENSOR_ID] = self._condition
-        attr[ATTR_SITE_ID] = self.site.id
-        attr[ATTR_SITE_NAME] = self.site.name
-        return attr
-
-    def update(self):
-        """Update current conditions."""
-        self.data.update()
-
-
-class MetOfficeCurrentData:
-    """Get data from Datapoint."""
-
-    def __init__(self, hass, datapoint, site):
-        """Initialize the data object."""
-        self._datapoint = datapoint
-        self._site = site
-        self.data = None
-
-    @Throttle(MIN_TIME_BETWEEN_UPDATES)
-    def update(self):
-        """Get the latest data from Datapoint."""
-        try:
-            forecast = self._datapoint.get_forecast_for_site(self._site.id, "3hourly")
-            self.data = forecast.now()
-        except (ValueError, dp.exceptions.APIException) as err:
-            _LOGGER.error("Check Met Office %s", err.args)
-            self.data = None
+        return {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            ATTR_LAST_UPDATE: self.metoffice_now.date if self.metoffice_now else None,
+            ATTR_SENSOR_ID: self._type,
+            ATTR_SITE_ID: self.metoffice_site_id if self.metoffice_site_id else None,
+            ATTR_SITE_NAME: self.metoffice_site_name
```

Comment:
Are the site attributes important for automations? We only want to add static info to state attributes if it's essential for automations.

Reply:
I don't use them myself - they were there from previous versions. I'll be honest, all I've done is bring the code in line with how I've updated the data model and UI/config_flow usage. If you don't want them there, then they can go? Which bits don't you want? Any of them?

Reply:
Happy to remove them, to be honest, and see if anyone complains?

Reply:
Let's keep them for now.